test/support/flux_parser.ex

```diff
@@ -0,0 +1,150 @@
+defmodule TelemetryInfluxDB.Test.FluxParser do
+  alias NimbleCSV.RFC4180, as: CSV
+
+  @column_types %{
+    "boolean" => :boolean,
+    "double" => :double,
+    "string" => :string,
+    "long" => :long,
+    "unsignedLong" => :unsigned_long,
+    "dateTime:RFC3339" => :datetime
+  }
+
+  def parse_tables(csv) do
+    csv
+    |> parse_chunks()
+    |> Enum.flat_map(fn chunk ->
+      table_data =
+        chunk
+        |> extract_table_text()
+        |> parse_csv()
+        |> separate_tables()
+
+      annotation_data =
+        chunk
+        |> extract_annotation_text()
+        |> parse_csv()
+
+      Enum.flat_map(table_data, fn table ->
+        case length(table) do
+          0 ->
+            %{}
+
+          _ ->
+            [column_names | table_rows] = table
+            column_types = annotation_data |> get_column_types()
+            parse_table(table_rows, column_names, column_types)
+        end
+      end)
+    end)
+  end
+
+  defp separate_tables(parsed) when parsed == [], do: [[]]
+
+  defp separate_tables([headers | rows]) do
+    table_index = Enum.find_index(headers, fn header -> header == "table" end)
+
+    rows
+    |> Enum.chunk_by(fn row -> Enum.at(row, table_index) end)
+    |> Enum.map(fn table_rows -> List.insert_at(table_rows, 0, headers) end)
+  end
+
+  def get_column_types(annotation_data) do
+    col_types_index =
+      annotation_data
+      |> Enum.find_index(fn a -> List.first(a) == "#datatype" end)
+
+    annotation_data
+    |> Enum.at(col_types_index)
+  end
+
+  defp parse_table(
+         table,
+         [_datatype | column_names],
+         [_ | column_types]
+       ) do
+    Enum.map(table, fn [_empty | row] -> parse_row(column_types, column_names, row) end)
+  end
+
+  defp parse_row(types, columns, row) do
```

Comment:
🙋‍♂ Might be worth making `row` the first param here to follow the Elixir convention of "the thing I'm operating on comes first".